src/test/java/plugins/DashboardViewPluginTest.java

```diff
@@ -45,6 +56,67 @@ public void configure_dashboard() {
     }
 
     @Test
+    public void jobsGridPortlet_fillColumnsFirst() throws MalformedURLException {
+        FreeStyleJob j1 = createFreeStyleJob();
+        FreeStyleJob j2 = createFreeStyleJob();
+        FreeStyleJob j3 = createFreeStyleJob();
+        FreeStyleJob j4 = createFreeStyleJob();
+
+        DashboardView v = createDashboardView();
+        JobsGridPortlet jobsGridPortlet = v.addBottomPortlet(JobsGridPortlet.class);
+        jobsGridPortlet.setNumberOfColumns(3);
+        jobsGridPortlet.setFillColumnFirst(true);
+        v.save();
+
+        assertThat(jobsGridPortlet.openJob(1, 3), nullValue());
+        assertThat(jobsGridPortlet.openJob(2, 2), notNullValue());
+
+        v.configure();
+        jobsGridPortlet.setFillColumnFirst(false);
+        v.save();
+        assertThat(jobsGridPortlet.openJob(1, 3), notNullValue());
+        assertThat(jobsGridPortlet.openJob(2, 2), nullValue());
+    }
+
+    @Test
+    public void jobsGridPortlet_numberOfColumns() throws MalformedURLException {
+        // One job is required for the portlet to be displayed
+        createFreeStyleJob();
+
+        DashboardView v = createDashboardView();
+        JobsGridPortlet jobsGridPortlet = v.addBottomPortlet(JobsGridPortlet.class);
+
+        jobsGridPortlet.setNumberOfColumns(2);
+        v.save();
+        assertThat(jobsGridPortlet.openJob(1, 2), nullValue());
+        assertJobsGridOutOfBounds(jobsGridPortlet, 1, 3);
+
+        v.configure();
+        jobsGridPortlet.setNumberOfColumns(3);
+        v.save();
+        assertThat(jobsGridPortlet.openJob(1, 3), nullValue());
+        assertJobsGridOutOfBounds(jobsGridPortlet, 1, 4);
+    }
+
+    /**
+     * Assert, that the given position in the given portlet does not exist.
+     *
+     * @param portlet to look into.
+     * @param row     to look for.
+     * @param column  to look for.
+     * @throws MalformedURLException
+     */
+    private void assertJobsGridOutOfBounds(JobsGridPortlet portlet, int row, int column) throws MalformedURLException {
```

Comment:
I did it like this on purpose. An expected exception would be active for the whole test. There might be another NoSuchElementException, that I do not want to ignore.